test/test_time_series.jl

```diff
@@ -219,6 +277,86 @@ end
     @test_throws IS.ConflictingInputsError IS.add_time_series!(sys, component, data)
 end
 
+@testset "Test add SingleTimeSeries with Polynomial Cost" begin
+    sys = IS.SystemData()
+    name = "Component1"
+    component = IS.TestComponent(name, 5)
+    IS.add_component!(sys, component)
+
+    initial_time = Dates.DateTime("2020-09-01")
+    resolution = Dates.Hour(1)
+    other_time = initial_time + resolution
+    polynomial_cost = repeat([(999.0, 1.0)], 365)
+    data = TimeSeries.TimeArray(
+        range(initial_time; length = 365, step = resolution),
+        polynomial_cost,
+    )
+    data = IS.SingleTimeSeries(data = data, name = "test_c")
+    IS.add_time_series!(sys, component, data)
+    ts1 = IS.get_time_series(
```

Comment:
Is there a reason why you aren't checking the returned data in this test and the next one?  You do check it in later tests.